tests/0102-static_group_rebalance.c

```diff
@@ -411,6 +446,276 @@ static void do_test_static_group_rebalance(void) {
+        c[1].expected_rb_event = RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS;
+        while (!static_member_wait_rebalance(&c[0], rebalance_start,
+                                             &c[0].assigned_at, 1000)) {
+                c[1].curr_line = __LINE__;
+                test_consumer_poll_once(c[1].rk, &mv, 0);
+        }
+
+        static_member_expect_rebalance(&c[1], rebalance_start,
+                                       &c[1].assigned_at, -1);
+
+        TEST_SAY("== Testing consumer unsubscribe ==\n");
+
+        /* Unsubscribe doesn't invoke a rebalance with static membership */
+
+        /* Send ConsumerGroupHeartbeat(-2), not leaving the group */
+        rebalance_start  = test_clock();
+        prev_assigned[0] = c[0].assigned_at;
+        prev_revoked[0]  = c[0].revoked_at;
+
+        prev_member_epoch[0] = consumer_generation_id(c[0].rk);
+        prev_member_epoch[1] = consumer_generation_id(c[1].rk);
+
+        rd_kafka_unsubscribe(c[1].rk);
+
+        /* Await revocation */
+        c[1].expected_rb_event = RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS;
+        static_member_expect_rebalance(&c[1], rebalance_start, &c[1].revoked_at,
+                                       -1);
+
+        rd_usleep(3000 * 1000, 0);
+        static_member_expect_no_rebalance(&c[0], prev_assigned[0],
+                                          prev_revoked[0]);
+
+        test_consumer_subscribe(c[1].rk, topics);
+
+        /* Await assignment */
+        c[1].expected_rb_event = RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS;
+        while (!static_member_wait_rebalance(&c[1], rebalance_start,
+                                             &c[1].assigned_at, 1000)) {
+                c[0].curr_line = __LINE__;
+                test_consumer_poll_once(c[0].rk, &mv, 0);
+        }
+
+        member_epoch[0] = consumer_generation_id(c[0].rk);
+        member_epoch[1] = consumer_generation_id(c[1].rk);
+
+        TEST_ASSERT(prev_member_epoch[0] == member_epoch[0],
+                    "c[0] should have the same member epoch "
+                    "as before the unsubscribe. "
+                    "expected %" PRId32 ", got %" PRId32,
+                    prev_member_epoch[0], member_epoch[0]);
+        TEST_ASSERT(prev_member_epoch[1] == member_epoch[1],
+                    "c[1] should have the same member epoch "
+                    "as before the unsubscribe. "
+                    "expected %" PRId32 ", got %" PRId32,
+                    prev_member_epoch[1], member_epoch[1]);
+
+        TEST_SAY("== Testing max poll violation ==\n");
+
+        /*
+         * Stop polling consumer 2 until we reach
+         * `max.poll.interval.ms` and is evicted from the group.
+         */
+        rebalance_start        = test_clock();
+        prev_assigned[0]       = c[0].assigned_at;
+        prev_revoked[0]        = c[0].revoked_at;
+        c[1].expected_rb_event = RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS;
+
+        for (i = 0; i < 10; i++) {
```

Comment:
Extract `max.poll.interval.ms` time to a variable `max_poll_interval_ms` and this for loop should be `for (i = 0; i < (max_poll_interval_ms/1000) + 1; i++)`